### ISSUE TITLE
[no-shadow] unknown compiler option 'no-shadow'

### ISSUE THREAD

**Issue by stefango** [typescript-eslint/typescript-eslint#4461]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": ["error"]
  }
}
```

```TS
// your repro code case
const foo;
[].map(foo=>foo); // ESLint: 'foo' is already declared in the upper scope.(@typescript-eslint/no-shadow)
```

**Expected Result**

Disable this rule as the [no-shadow doc](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-shadow.md) says

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.1` |
| `@typescript-eslint/parser`        | `4.14.1` |
| `TypeScript`                       | `4.5.3` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `16.13.1` |

**Comment by JoshuaKGoldberg** [comment]
@stefango please fill out the issue template completely. We can't tell from just what you posted what your issue is. The rule seems to be working correctly in the snippet.

**Comment by stefango** [comment]
> @stefango please fill out the issue template completely. We can't tell from just what you posted what your issue is. The rule seems to be working correctly in the snippet.

Issue review plz. I have filled out it completely.

**Comment by JoshuaKGoldberg** [comment]
The error message you're seeing is from the typescript-eslint rule:

> (@typescript-eslint/no-shadow)

There are two "no-shadow" rules: the "base" rule that comes from ESLint and the "extension" rule provided as part of typescript-eslint. Disabling the base `no-shadow` rule means only the base rule is disabled so you don't get duplicate rule complaints from both rules.

Your config specifies that the extension rule should give errors:

```json
    "@typescript-eslint/no-shadow": ["error"]
```

**Comment by stefango** [comment]
> > @stefango please fill out the issue template completely. We can't tell from just what you posted what your issue is. The rule seems to be working correctly in the snippet.
> 
> Issue review plz. I have filled out it completely.

Sorry for my mistake on filling out Expected/Actual Result. `unknown compiler option 'no-shadow'` error should not turn out in Webstorm when I disabled this rule as the [no-shadow doc](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-shadow.md) says.

Seems like tsconfig haven't support no-shadow option?
https://www.typescriptlang.org/tsconfig#compilerOptions

**Comment by bradzacher** [comment]
please have a read through our docs on setting up your linting
